api/collections/views.py

```diff
@@ -0,0 +1,246 @@
+    def get_object(self):
+        smart_folders = (
+            '~amr',
+            '~amp',
+        )
+
+        node_id = self.kwargs[self.node_lookup_url_kwarg]
+        for folder_id in smart_folders:
+            if node_id == folder_id:
+                node = Node.find(node_id)
+                node.smart_folder = True
+                smart_folder_node = {
+                    'id': node_id,
+                    'title': node.title,
+                    'date_created': node.date_created,
+                    'date_modified': node.date_modified,
+                    'properties': {
+                        'smart_folder': True,
+                        'dashboard': node.is_dashboard,
+                        'collection': node.is_folder,
+                    },
+                }
+                return smart_folder_node
+
+        node = self.get_node()
+        node.smart_folder = False
+        return node
+
+    # overrides RetrieveUpdateAPIView
+    def get_serializer_context(self):
+        # Serializer needs the request in order to make an update to privacy
+        return {'request': self.request}
+
+    # overrides DestroyAPIView
+    def perform_destroy(self, instance):
+        user = self.request.user
+        auth = Auth(user)
+        node = self.get_node()
+        date = datetime.now()
+        # Right now you cannot DELETE the dashboard but if you try, you get a 204
+        if not node.is_dashboard:
+            node.remove_node(auth, date)
+        node.save()
+
+
+class CollectionChildrenList(generics.ListAPIView, CollectionMixin):
+    """Children of the current collection.
+
+    This will get the next level of child nodes for the selected collection if the current user has read access for those
+    nodes. Currently, if there is a discrepancy between the children count and the number of children returned, it
+    probably indicates private nodes that aren't being returned. That discrepancy should disappear before everything
+    is finalized.
+    """
+    permission_classes = (
+        drf_permissions.IsAuthenticated,
+        ContributorOrPublic
+    )
+
+    serializer_class = CollectionSerializer
+
+    # overrides ListAPIView
+    def get_queryset(self):
+        nodes = self.get_node().nodes
+        user = self.request.user
+        if user.is_anonymous():
+            auth = Auth(None)
+        else:
+            auth = Auth(user)
+        children = [node for node in nodes if node.can_view(auth)]
+        return children
+
+
+class CollectionPointersList(generics.ListCreateAPIView, CollectionMixin):
+    """Pointers to other nodes.
+
+    Pointers are essentially aliases or symlinks: All they do is point to another node.
+    """
+    permission_classes = (
+        drf_permissions.IsAuthenticated,
+        ContributorOrPublic,
+    )
+
+    serializer_class = CollectionPointersSerializer
+
+    def get_queryset(self):
+        pointers = self.get_node().nodes_pointer
```

Comment:
It looks like we aren't actually performing any of the queries for the smart folders. We'll have to check to see if this is a smart folder, and then we'll need to do the smart folder query here if it is. That will likely mean moving the list of smart folders to somewhere that all the classes can use and turning it into a dictionary so that we can store the query along with the smart folders themselves.